### ISSUE TITLE
feature request: debug log traces to be monitored in console

### ISSUE THREAD

**Issue by kirisakow** [gildas-lormeau/SingleFile#1475]
### Is your feature request related to a problem? Please describe.

no log traces to monitor in console

### Describe the solution you'd like

implement log traces to be monitored in console

### Describe alternatives you've considered (optional)

NA

### Additional context (optional)

I often use SingleFile in Firefox on Android device. Hence two problems:

a. On certain websites (I'm not saying SingleFile's to blame) the saving process gets systematically stuck on `Step 2 / 3` in what seems like a never-ending loop.

b. ...with no log traces to monitor in console to help me understand with what's wrong.

**Comment by gildas-lormeau** [comment]
Since the problem is related to Android, I understand the need. Normally, SingleFile never gets stuck on step 2/3 but you may have to wait a few minutes. This is due to pending connections.  You can activate the `Network > set maximum download time` option if this problem occurs too often.

**Comment by kirisakow** [comment]
> You can activate the `Network > set maximum download time` option if this problem occurs too often.

So, I did that, on the same websites on which I usually use SingleFile: I set network timeout to 5s — **unfortunately to no effect** (ie it didn't help). 

Believe it or not, what happens in SingleFile in Firefox **_for Android_** really looks like an infinite loop. And it happens way more than half of the time. There really must be something that's incompatible with Firefox for Android. And the most efficient way to know what it is — is to add logging. Please 🙏 Thanks in advance.

**Comment by gildas-lormeau** [comment]
To do some tests, do you know of a website where this problem occurs?

**Comment by kirisakow** [comment]
[This one](https://www.arretsurimages.net/chroniques/obsessions/soiree-dialectique-a-montreuil) may be a good example!

**Comment by gildas-lormeau** [comment]
Unfortunately, I was not able to reproduce the issue. However, I encountered a similar problem when downloading a very large file in ZIP format. So I applied the same fix during step 2/3. I hope this will correct the problem. The fix will be available in the next version.

**Comment by gildas-lormeau** [comment]
@kirisakow Are you still suffering from this issue?

**Comment by kirisakow** [comment]
Yes, I do. I am still experiencing an “infinite loop” effect in Firefox for Android, and still no hint or details on screen about what's going on.

I'd really love to help to debug this. I've tried to USB-debug it via ADB, but I couldn't extract any logs or anything meaningful.

**Comment by gildas-lormeau** [comment]
@ kirisakow I noticed a similar behavior when the device has some RAM issues. It looks like Firefox can kill the extension in such a case. How much RAM do you have?

**Comment by kirisakow** [comment]
4GB RAM

Firefox killing an extension's process may be a possible reason indeed. I also noticed SingleFile often works well in Android in a freshly started Firefox runtime, whereas it almost never works in a Firefox that has been running for a while.

**Comment by gildas-lormeau** [comment]
@kirisakow Thanks for the info, I think it's definitely something to look into in this case.
